lib/Sema/TypeCheckAttr.cpp

```diff
@@ -4581,6 +4583,42 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
   (void)attr->getParameterIndices();
 }
 
+/// Checks if original candidate and registered derivative match in terms
```

Comment:
Do you think there is a better place to move this utility functions to?

Reply:
The treatment of constructors as static functions for the purpose of differentiability witness matching is pretty unique. I think this is where it needs to be.

Reply:
ok, great! thanks.